### PR TITLE
add test for tcms.rpc.api.attachment

### PR DESCRIPTION
**Pull request by awalvie**

### CODE REVIEW

File: tcms/rpc/tests/test_attachment.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=attribute-defined-outside-init, invalid-name, objects-update-used
+
+from tcms.rpc.tests.utils import APITestCase
+from tcms.tests.factories import TestPlanFactory
+
+class TestRemoveAttachment(APITestCase):
```
Comment (atodorov) [inline tcms/rpc/tests/test_attachment.py:7]: Missing test class for permissions

File: tcms/rpc/tests/test_attachment.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.plan = TestPlanFactory()
+
+    def test_remove_attachment(self):
+        file_name = 'attachment.txt'
+
+        self.rpc_client.TestPlan.add_attachment(self.plan.pk, file_name, 'a2l3aXRjbXM=')
+        attachments = self.rpc_client.TestPlan.list_attachments(self.plan.pk)
```
Comment (atodorov) [inline tcms/rpc/tests/test_attachment.py:19]: In this scenario that would happen during setup and then just try to remove the file and verify it's not there anymore.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/rpc/tests/test_attachment.py (hunk 1 above)
Comment (awalvie) [reply tcms/rpc/tests/test_attachment.py:7]: Yeah, about that. I originally wanted to have the permissions test as well, but I've been running into some issues. I would need the permissions for `attachments.add_attachment` to add an attachment as well as `attachments.remove_attachment` to remove it, but I don't see a way to have both permissions simultaneously.

Comment (awalvie) [reply tcms/rpc/tests/test_attachment.py:7]: Unless I'm approaching it from the wrong angle.

Comment (atodorov) [reply tcms/rpc/tests/test_attachment.py:7]: Option 1) use utils.add_attachment() directly (see TestPlan.add_attachment method) - it calls another view which takes care of the actual operation and it only needs the user to be logged in.

Option 2) use user_should_have_perm() during setup to grant the additional permissions for adding and listing attachments.See how it is used in existing tests.

**Pull request merged**